lib/qam.pm

```diff
@@ -73,31 +73,42 @@ sub is_patch_needed {
     }
 }
 
+sub add_repo_if_not_present {
+    my ($url, $name) = @_;
+    my $gpg = get_var('BUILD') =~ m/^MR:/ ? "-G" : "";
+    my $system_repos = script_output('zypper lr -u');
+    unless ($system_repos =~ /$url/) {
+        zypper_call("--no-gpg-checks ar -f $gpg -n '$name' $url '$name'");
+    }
+}
+
+# https://progress.opensuse.org/issues/90522
+sub add_extra_customer_repositories {
+    my $arch = get_var('ARCH');
+
+    if (is_sle('=12-SP2')) {
+        add_repo_if_not_present("http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP2-LTSS-ERICSSON/$arch/update/", '12-SP2-LTSS-ERICSSON-Updates');
+    }
+    elsif (is_sle('=12-SP3')) {
+        add_repo_if_not_present("http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP3-LTSS-TERADATA/$arch/update/", '12-SP3-LTSS-TERADATA-Updates');
+    }
+    elsif (is_sle('=15-SP3')) {
+        add_repo_if_not_present("http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP3-ERICSSON/$arch/update/", '15-SP3-ERICSSON-Updates');
+    }
+    elsif (is_sle('=15-SP4')) {
+        add_repo_if_not_present("http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP4-ERICSSON/$arch/update/", '15-SP4-ERICSSON-Updates');
```

Comment:
When there are so many repos, it'd be cleaner to define an array and loop over it:

    my @repo_list = (
        {cond => '=12-SP2', name => '12-SP2-LTSS-ERICSSON-Updates', uri => '...'},
        {...}
    );
    
    for my $repo in (@repo_list) {
        add_repo_if_not_present($repo->{uri}, $repo->{name}) if is_sle($repo->{cond});
    }

Reply:
I had feeling to make it better, this is great. Thank you